### ISSUE TITLE
Overflow on Winpython in make_tensor_proto in tensor_util.py

### ISSUE THREAD

**Issue by ghostbust555** [fchollet/keras#5965]
in the method make_tensor_proto there is a call to
shape_size = np.prod(shape)

which is overflowing and returning a negative number even on 64 bit windows/numpy
replacing that line with 

shape_size = reduce(mul, shape, 1)

has solved my overflow problems.
This only appeared to happen when using a fit generator which is very strange.
I am using latest keras, latest tensorflow on windows x64

my model was 
 
input is 250x250x3

    x = Conv2D(filters=32, kernel_size=3, strides=2, padding='valid')(input)
    x = Activation("relu")(x)

    x = Dropout(.2)(x)

    x = Flatten()(x)

    x = Dense(nb_output, activation='softmax')(x)

    model = Model(inputs=input, outputs=x)
    model.compile(loss='categorical_crossentropy',
                       optimizer='adam',
                       metrics=['accuracy'])

model.fit_generator(generate_arrays_from_file(), steps_per_epoch=(trainSize/batchSize), epochs=100, verbose=1)
